src/libraries/System.Formats.Tar/src/System/Formats/Tar/TarHeader.Read.cs

```diff
@@ -20,15 +20,15 @@ internal sealed partial class TarHeader
         // Attempts to retrieve the next header from the specified tar archive stream.
         // Throws if end of stream is reached or if any data type conversion fails.
         // Returns a valid TarHeader object if the attributes were read successfully, null otherwise.
-        internal static TarHeader? TryGetNextHeader(Stream archiveStream, bool copyData, TarEntryFormat initialFormat)
+        internal static TarHeader? TryGetNextHeader(Stream archiveStream, bool copyData, TarEntryFormat initialFormat, bool processDataBlock)
         {
             // The four supported formats have a header that fits in the default record size
             Span<byte> buffer = stackalloc byte[TarHelpers.RecordSize];
 
             archiveStream.ReadExactly(buffer);
 
             TarHeader? header = TryReadAttributes(initialFormat, buffer);
-            if (header != null)
+            if (header != null && processDataBlock)
```

Comment:
?
```suggestion
            if (header != null && initialFormat != TarEntryFormat.Pax)
```

Reply:
Good question. That was my initial approach, but then I opted for the extra boolean for these reasons:

- We have multiple types of PAX entries: Global, Extended Attributes, and normal entries. I was introducing regressions in some of those cases.
- It's much more clear what the boolean is for, rather than a not-equal check.